src/foam/demos/u2/Border.js

```diff
@@ -619,3 +619,100 @@ foam.CLASS({
     }
   ]
 });
+
+foam.CLASS({
+  name: 'ContextSwitchBorder',
+  extends: 'foam.u2.Element',
+
+  documentation: `
+    An unstyled border. Intended for use as a default value for
+    border properties.
+  `,
+
+  // exports: ['controllerMode', 'notCM'],
+  properties: [
+    {
+      name: 'controllerMode',
+      value: foam.u2.ControllerMode.VIEW
+    }
+  ],
+
+  methods: [
+    function render() {
+      this
+      .start()
+      .startContext({ controllerMode: foam.u2.ControllerMode.EDIT, data: this })
+      .add('Added from Border')
+      .add(this.CONTROLLER_MODE.__)
+      .call(function() { console.log('*******Border', this.controllerMode, this.__context__.controllerMode, this.__subSubContext__.controllerMode)})
```

Comment:
Probably should remove htis debug log before merging

Reply:
This is demo code, so console logs help with demoing. If moved out of demos, then yes.